addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js

```diff
@@ -94,22 +94,42 @@ export class ProductScreen extends Component {
             ? [...selectedCategory.child_ids]
             : this.pos.models["pos.category"].filter((category) => !category.parent_id);
     }
+
+    getSelectedChildren() {
+        const selectedCategoryAllParentsIds = (this.pos.selectedCategory?.allParents || [])
+            .map((category) => category.id)
+            .concat([this.pos.selectedCategory?.id]);
+        return this.pos.models["pos.category"].flatMap((category) =>
+            selectedCategoryAllParentsIds.includes(category.id) ? category.child_ids : []
+        );
```

Comment:
```suggestion
        const selected = this.pos.selectedCategory ? [this.pos.selectedCategory] : [];
        const allParents = (this.pos.selectedCategory?.allParents || []).concat(selected);
        return allParents.flatMap(category => category.child_ids);
```

Can you try this?

Reply:
@caburj correct me if i'm wrong, we should filter on `selectedCategoryAllParentsIds` no ? the behavior is to show only the child categories if the parent is selected.
Wdyt ?

Reply:
For the moment, I don't disagree with your solution. My suggestion is just a refactoring of you current change.

Reply:
Sorry, yes makes sense. applied thank you.